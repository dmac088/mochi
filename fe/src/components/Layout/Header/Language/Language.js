import React from "react";

const changeLang = (e, props) => {
    e.preventDefault();
    console.log(props);
    props.history.push('/' + e.currentTarget.id + '/HKD');
}

export const Language = (props) => {
    return (
        <li> <a href="#">English <i className="fa fa-chevron-down"></i></a>
            <ul>
                <li><a id="en-GB" onClick={(e) => changeLang(e, props)}>English</a></li>
                <li><a id="zh-HK" onClick={(e) => changeLang(e, props)}>Chinese</a></li>
            </ul>
        </li>
    );
}

export default Language;






