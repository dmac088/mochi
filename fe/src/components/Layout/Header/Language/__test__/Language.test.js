import React from 'react';
import ReactDOM from 'react-dom';
import { LanguageBase } from '../Language';
import { BrowserRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history'
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });

describe('Test language menu', () => {

    afterEach(cleanup);
    
 
    it("renders without crashing", () => {
        const div = document.createElement("div"); 
        const historyMock = { history: { push: jest.fn() }}
        ReactDOM.render(
            <BrowserRouter>
                <LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock} }/>
            </BrowserRouter >, div);
    });
 
    it('should change the router params to zh-HK when menu item is clicked', () => {
        //mock a history object with mock function
        const historyMock = { history: { push: jest.fn() }}
        //create a shallow copy of the component we want to render and mock the props
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock}} />);
        //find the element we want to click
        const link = wrapper.find('a#zh-HK');
        //click the element and mock additional function params
        link.simulate('click', {
            preventDefault: () => {},
            currentTarget: { id: 'zh-HK' }
        });
        //get the expected urlString
        const urlString = "/zh-HK/HKD/"
        //assert that the expected result is the expected string
        expect(historyMock.history.push.mock.calls[0][0]).toEqual(urlString);
    });

    it('should change the router params to en-GB when menu item is clicked', () => {
        //mock a history object with mock function
        const historyMock = { history: { push: jest.fn() }}
        //create a shallow copy of the component we want to render and mock the props
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock}} />);
        //find the element we want to click
        const link = wrapper.find('a#en-GB');
        //click the element and mock additional function params
        link.simulate('click', {
            preventDefault: () => {},
            currentTarget: { id: 'en-GB' }
        });
        //get the expected urlString
        const urlString = "/en-GB/HKD/"
        //assert that the expected result is the expected string
        expect(historyMock.history.push.mock.calls[0][0]).toEqual(urlString);
    });

    it('renders English menu item correctly', () => {
        const historyMock = { history: { push: jest.fn() }}
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock}} />);
        const text = wrapper.find('a#en-GB');
        expect(text.text()).toBe('English');
    });

    it('renders Chinese menu item correctly', () => {
        const historyMock = { history: { push: jest.fn() }}
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock}} />);
        const text = wrapper.find('a#zh-HK');
        expect(text.text()).toBe('Chinese');
    });
    
    it("renders english in language menu header correctly", () => {
        const historyMock = { history: { push: jest.fn() }}
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "en-GB", curr: "HKD" }, ...historyMock}} />);
        const text = wrapper.find('a#language');
        expect(text.text()).toBe("en-GB");
    });

    it("renders chinese in language menu header correctly", () => {
        const historyMock = { history: { push: jest.fn() }}
        const wrapper = shallow(<LanguageBase {...{ params: { lang: "zh-HK", curr: "HKD" }, ...historyMock}} />);
        const text = wrapper.find('a#language');
        expect(text.text()).toBe("zh-HK");
    });

    it("matches snapshot 1", () => {
        const history = createMemoryHistory('/en-GB/HKD') 
        const tree = renderer.create(
            <BrowserRouter>
                <LanguageBase
                    { ...{ params: { lang: "en-GB", curr: "HKD" }}}
                    history={history} />
            </BrowserRouter >
        ).toJSON();
        expect(tree).toMatchSnapshot();
    }); 

}); 