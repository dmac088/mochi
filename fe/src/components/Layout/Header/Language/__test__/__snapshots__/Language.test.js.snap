// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  data-testid="language"
>
   
  <a
    href="#"
  >
     
    <i
      className="fa fa-chevron-down"
    />
  </a>
  <ul>
    <li>
      <a
        href="#"
        id="en-GB"
        onClick={[Function]}
      >
        English
      </a>
    </li>
    <li>
      <a
        href="#"
        id="zh-HK"
        onClick={[Function]}
      >
        Chinese
      </a>
    </li>
  </ul>
</li>
`;
